x-pack/plugins/spaces/server/lib/request_interceptors/on_post_auth_interceptor.test.ts

```diff
@@ -109,45 +163,48 @@ describe('onPostAuthRequestInterceptor', () => {
         },
       };
 
+      const log = {
+        log: jest.fn(),
+        trace: jest.fn(),
+        debug: jest.fn(),
+        info: jest.fn(),
+        warn: jest.fn(),
+        error: jest.fn(),
+        fatal: jest.fn(),
+      };
+
       let basePath: string | undefined;
-      server.decorate('request', 'getBasePath', () => basePath);
-      server.decorate('request', 'setBasePath', (newPath: string) => {
-        basePath = newPath;
-      });
 
-      // The onRequest interceptor is also included here because the onPostAuth interceptor requires the onRequest
-      // interceptor to parse out the space id and rewrite the request's URL. Rather than duplicating that logic,
-      // we are including the already tested interceptor here in the test chain.
-      initSpacesOnRequestInterceptor(server);
-      initSpacesOnPostAuthRequestInterceptor(server);
+      const httpMock = httpServiceMock.createSetupContract();
 
-      server.route([
-        {
-          method: 'GET',
-          path: '/',
-          handler: (req: any) => {
-            return { path: req.path, url: req.url, basePath: req.getBasePath() };
-          },
-        },
-        {
-          method: 'GET',
-          path: '/app/{appId}',
-          handler: (req: any) => {
-            return { path: req.path, url: req.url, basePath: req.getBasePath() };
-          },
-        },
-        {
-          method: 'GET',
-          path: '/api/foo',
-          handler: (req: any) => {
-            return { path: req.path, url: req.url, basePath: req.getBasePath() };
-          },
-        },
-      ]);
+      httpMock.basePath.get = jest.fn().mockImplementation(() => basePath);
+      httpMock.basePath.set = jest.fn().mockImplementation((req: any, newPath: string) => {
+        basePath = newPath;
+      });
+      httpMock.registerOnPreAuth = jest.fn().mockImplementation(async handler => {
+        const preAuthRequest = {
```

Comment:
JFYI there is a hapi request mock, which is not exported outside yet. do you think we should provide `KibanaRequest` mock and export it as well? https://github.com/restrry/kibana/blob/2f22c48dbaec368b3e43a4a30bb50dd369b8e463/src/core/server/http/http_server.mocks.ts#L55